### ISSUE TITLE
DPL-428 - As developers we need to analyse the amount of work needed to upgrade SMRTLink to v11 and generate the stories

### ISSUE THREAD

**Issue by stevieing** [sanger/traction-service#777]
**Description**
As developers we need to analyse the amount of work needed to upgrade SMRTLink to v11 and generate the stories

**Who the primary contacts are for this work**
James W / Craig

**Knowledge or Stake holders**
Other people that may have specific knowledge about this work or have a stake in how it is implemented. e.g. John Smith is an expert on x

**Additional context or information**

- Problem with barcode file installed into SLv11 - We see this as a process issue rather than a bug so maybe we need to align the tag sets used in UAT and production. I will catch up with Craig and maybe Carol next week.
- Redundant: Removing Generate HiFi Reads - Although this field is redundant it does not prevent the file being uploaded. It includes a database column so PSD need to decide how best to remove it.
- Bug: Imported barcoded libraries not recognised as barcoded. We need to add new Header Demultipex Barcodes. This does not require database changes. Required for upgrade to SMRT Link 11
- New SLv11 Run Design Options - These are required for MVP. The minimum would be to add these as columns with defaults. Long Read would need the extra functionality but this would not be required for initial move to SMRT Link v11
- Set Sequencing application in sample sheet based on library type - James has already done some preparatory work for this but this will require some planning as it is complex

Next Steps:

- James G and Steve to write up stories and then we will pass them on for review. We should be able to get the MVP stories prioritised next week.
- James G and Steve to research how to add versioning which will make future updates easier.
- James G and Steve to research how to add a single column to cope with database entries for sample sheets. Again this should make future updates easier

**Comment by stevieing** [comment]
I have gone through Traction UAT version and UAT SMRT Link 11. There are a few incompatibilities and I have listed them in the attached pdf but there is a live google slide doc here:

https://docs.google.com/presentation/d/1Pyg2p1_8CICC4RGU4TAJYYSIIh8bEd7xQPV1JhOd5N0/edit?usp=sharing

 

The fixes and requests on pages 2,4 & 5 would be considered essential for our roll out of SLv11 to the production server.

 

If it helps, I have attached 3x sample sheets related to TRACTION-RUN-117:

TRACTION-RUN-117.csv is the file currently generated by UAT Traction

TRACTION-RUN-117_barcodecorrect_bcdeplex.csv is the file required to be compatible with the current install of uat SL11 (i.e. I’ve manually edited the tag names to successfully import and edited the header and options for barcode deplexing)

TRACTION-RUN-117_barcodecorrect_bcdeplex_extras.csv is the file with the extra fields from slide 5 that I have been manually altering to bug hunt with option switches.

 

NB the 2x altered csv files may not truly represent the run as currently set up in Traction run 117 now as I was using them for additional bug finding with the new options required.

 

Can this Traction work request be flagged as priority when scheduled?

 

Kind regards,

Craig
